src/org/opendatakit/briefcase/export/CipherFactory.java

```diff
@@ -0,0 +1,119 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.opendatakit.briefcase.export;
+
+import java.io.UnsupportedEncodingException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.apache.commons.codec.binary.Base64;
+import org.opendatakit.briefcase.model.CryptoException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles the initialization of Cipher objects for the decryption of submission
+ * and media files.
+ */
+final class CipherFactory {
+  private static final Logger log = LoggerFactory.getLogger(CipherFactory.class);
+  private static final int IV_BYTE_LENGTH = 16;
+  private final SecretKeySpec symmetricKey;
+  private final byte[] ivSeedArray;
+  private int ivCounter = 0;
+
+  private CipherFactory(String instanceId, byte[] symmetricKeyBytes) throws CryptoException {
+    symmetricKey = new SecretKeySpec(symmetricKeyBytes, "AES/CFB/PKCS5Padding");
+    try {
+      MessageDigest md = MessageDigest.getInstance("MD5");
+      md.update(instanceId.getBytes("UTF-8"));
+      md.update(symmetricKeyBytes);
+      byte[] messageDigest = md.digest();
+      ivSeedArray = new byte[IV_BYTE_LENGTH];
+      for (int i = 0; i < IV_BYTE_LENGTH; ++i) {
+        ivSeedArray[i] = messageDigest[(i % messageDigest.length)];
+      }
+    } catch (NoSuchAlgorithmException | UnsupportedEncodingException e) {
+      String msg = "Error constructing ivSeedArray";
+      log.error(msg, e);
+      throw new CryptoException(msg + " Cause: " + e);
+    }
+  }
+
+  /**
+   * Factory that initializes a new {@link CipherFactory} for the given instance ID,
+   * encryption key and private key values.
+   *
+   * @throws CryptoException
+   */
+  static CipherFactory from(String instanceId, String base64EncryptedKey, PrivateKey privateKey) {
+    try {
+      Cipher pkCipher;
+      pkCipher = Cipher.getInstance("RSA/NONE/OAEPWithSHA256AndMGF1Padding");
+      pkCipher.init(Cipher.DECRYPT_MODE, privateKey);
+      byte[] encryptedSymmetricKey = Base64.decodeBase64(base64EncryptedKey);
+      byte[] decryptedKey = pkCipher.doFinal(encryptedSymmetricKey);
+      return new CipherFactory(instanceId, decryptedKey);
```

Comment:
11:38:52.334 [main] ERROR org.opendatakit.briefcase.util.FileSystemUtils - Error decrypting:1524644764247.jpg.enc
java.security.InvalidKeyException: Illegal key size
⋮
	at org.opendatakit.briefcase.util.CipherFactory.getCipher(CipherFactory.java:78)
	at org.opendatakit.briefcase.util.EncryptionInformation.getCipher(EncryptionInformation.java:59)
	at org.opendatakit.briefcase.util.FileSystemUtils.decryptFile(FileSystemUtils.java:455)

Reply:
Can you confirm that you're using Java 8u161 or greater?

Reply:
JRE: 1.8.0_152-release-1136-b20 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.12.6

Reply:
OK, with that JVM you should follow these complicated and boring instructions here: https://docs.opendatakit.org/encrypted-forms/#configuration

**OR**

Install JVM 1.8.0u161 or a greater version :)